### ISSUE TITLE
Reloading gets you to latest toot, losing hability to read oldest unread first

### ISSUE THREAD

**Issue by ivangjxyz** [mastodon/mastodon-android#70]
Maybe this is a feature, not a bug of the app, but certainly it is what keeps me using Fedilab for now.

With Fedilab, the loading of new toots keeps my current position on the timeline, so I can read things in their natural order. In official Mastodon app, as it takes you to the top, final toot, it forces you to scroll back down to whatever was the last toot you saw.

**Comment by tjkirch** [comment]
I agree, this is the one feature I couldn't live without when trying the app.  It seems to discourage following many people because it's so hard to keep track of what you've read.

**Comment by oufmilo** [comment]
Run the last version 👍

**Comment by tjkirch** [comment]
This is working for me now, thank you! Can this (and the duplicate) be closed now, or is there more to finish?

**Comment by dtantsur** [comment]
Hmm, I've just tried the latest version, and got sent to the top on refresh. Maybe it regressed again?

**Comment by CubicrootXYZ** [comment]
> Hmm, I've just tried the latest version, and got sent to the top on refresh. Maybe it regressed again?

Can confirm, same here in v1.1.3, worked previously.

**Comment by garrett** [comment]
It seems to work below a certain post threshold. But once the "load more" button is visible, it loads and jumps to the top of the timeline.

If you're quick enough, you can tap and slide the timeline so the button is off the top of the screen and hope that it keeps your position, but that'd doesn't always work. The app _should_ just work by default and keep your place, no matter what.

I've been using Mastodon for a few years and do appreciate this app. It usually worked fine, as I didn't get as many posts... until the Musk takeover of Twitter. The more people I have on the app, the more posts show up... and that means there's almost always a "load more" button. Therefore, the app doesn't keep its position as much as it used to. It's probably the same for everyone who is now following a lot of people.

FWIW: I've tried a number of apps over the past few days as a result and Fedilab (free on F-Droid, but also for pay on the Google Play Store) is the only one that seems to keep a timeline position. Sadly, it's not as polished as the Mastodon app (although it does have more features).

This lack of reliable timeline remembering is really the main thing that's now keeping me away from the official Mastodon app, sadly.

---

TL;DR: Influx of new users means the loss of timeline position bug is triggered more frequently, as it seems to be related to the number of new posts (and the load more button).

**Comment by C-Duv** [comment]
I have the same issue (on iOS) and was used to _Twitterrific_ ability to keep track of my "feed consumption", a Google search led me here to discover the issue is quite real, on both iOS and Android :'(

Note: here is  similar issue on the iOS application (https://github.com/mastodon/mastodon-ios/issues/744) about switching tab.

**Comment by tjkirch** [comment]
@grishka Am I understanding correctly that a fix for alt text also somehow fixed this issue?

**Comment by grishka** [comment]
How did that happen?

**Comment by grishka** [comment]
Ah I see. The commit was cherrypicked from another repo where #70 is a relevant issue.
![image](https://github.com/mastodon/mastodon-android/assets/1478258/0fbee67b-d8fd-4e98-9684-a77ebffe2bfb)
That's messed up.

**Comment by tinsukE** [comment]
There are two situations I'm aware of that would cause the Home TL to jump to the top (or make you lose your reading position).

1. When tapping "Load missing posts", the missing posts would get loaded on a top-to-bottom fashion, so you'd have to scroll down to where you were previously reading so you could continue your chronological TL reading. This would be fixed by merging this PR: https://github.com/mastodon/mastodon-android/pull/680
2. When pulling to refresh. This is a tricky one. I thought fixing this would be a matter of just remembering the reading position and making sure to restore it after the refresh is done, but it is more complicated than that, because the current implementation actually discards the data for your current TL and replaces with what was fetched. IF your reading position wasn't too old, restoring it would still work, but there is no guarantee that it would. This is the code with this behavior:
https://github.com/grishka/appkit/blob/master/appkit/src/main/java/me/grishka/appkit/fragments/BaseRecyclerFragment.java#L320-L324
